src/foam/nanos/auth/CachedAuthServiceProxy.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * @license
+ * Copyright 2018 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.auth',
+  name: 'CachedAuthServiceProxy',
+  extends: 'foam.nanos.auth.ProxyAuthService',
```

Comment:
I think this should be a decorator rather than extending ProxyAuthService.

Reply:
Isn't extending a proxy a means to create a decorator? This way, I don't need to worry about any other methods on the interface working because they'll automatically get passed through to the delegate.

Reply:
Sorry, my mistake. I read Stub instead of Proxy somehow.